docker/docker.go

```diff
@@ -120,6 +120,10 @@ func (r *dockerRunner) loadConfig() (conf *dockerConfig, err error) {
 }
 
 func (r *dockerRunner) run(conf *dockerConfig) error {
+	if r.task.Plan.DryRun {
+		return nil
+	}
+
```

Comment:
We can move this logic to `ExecPlan`. In `ExecPlan.run`, we can return success instead of calling the `runner`. Then we avoid modifying every exec-driver.